### PR TITLE
moving over files from other repositories

### PR DESCRIPTION
**Pull request by mac-arrap**
Signed-off-by: Camilo Parra <caeparra@gmail.com>

### CODE REVIEW

**Comment by mikelodder7** [comment]
Did you mean to include `.pyc` files? I believe these should be removed.

**Comment by mac-arrap** [comment]
@mikelodder7 I have removed the files and will fix the dockerfile

**Comment by mac-arrap** [comment]
fixed

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      sudo \
+      nano \
+      git \
+      cmake \
+      autoconf \
+      libtool \
+      pkg-config \
+      libssl-dev
+
+ENV RUST_VERSION=${RUST_VERSION:-1.34.0}
```
Comment (mikelodder7) [inline docker/Dockerfile]: Update to rust 1.36

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --no-modify-path --default-toolchain $RUST_VERSION \
+    && chmod -R a+w $RUSTUP_HOME $CARGO_HOME \
+    && rustup --version \
+    && cargo --version \
+    && rustc --version
+
+WORKDIR /home/
+
+RUN curl -fsSL https://github.com/jedisct1/libsodium/releases/download/1.0.14/libsodium-1.0.14.tar.gz | tar -xz
```
Comment (mikelodder7) [inline docker/Dockerfile]: Is this needed? If so, please update to 1.0.17

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+WORKDIR libsodium-1.0.14
```
Comment (mikelodder7) [inline docker/Dockerfile]: Same as above

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+RUN ./autogen.sh && ./configure && make && sudo make install
+
+ENV SODIUM_LIB_DIR=/usr/local/lib \
```
Comment (mikelodder7) [inline docker/Dockerfile]: Ursa doesn't need libsodium

File: ursa/indy_error.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    CommonInvalidParam12 = 111,
+
+    # Invalid library state was detected in runtime. It signals library bug
+    CommonInvalidState = 112,
+
+    # Object (group, key, point, and etc...) passed by library caller has invalid structure
+    CommonInvalidStructure = 113,
+
+
+class IndyCryptoError(Exception):
```
Comment (lovesh) [inline ursa/indy_error.py]: Is it intentionally called `IndyCryptoError` and not `UrsaError` or something like that?

File: tests/bls-indy/conftest.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
+from ursa.bls import Bls, Generator, SignKey, VerKey, ProofOfPossession, Signature, MultiSignature
```
Comment (lovesh) [inline tests/bls-indy/conftest.py]: Is the directory intentionally called `bls-indy`?

File: ursa/indy_error.py (hunk 5 above)
Comment (mac-arrap) [reply ursa/indy_error.py]: this messaging type is specific to Indy... anyone is welcome to bring in their own error handler

**Pull request merged**